imap_processing/lo/l0/lol0.py

```diff
@@ -0,0 +1,48 @@
+"""General Lo L0 data class used for parsing data and setting attributes."""
+from dataclasses import dataclass, fields
+
+from imap_processing.ccsds.ccsds_data import CcsdsData
+
+
+@dataclass
+class LoL0:
+    """Data structure for common values across histogram and direct events data.
+
+    Attributes
+    ----------
+    ground_sw_version : str
+        Ground software version
+    packet_file_name : str
+        File name of the source packet
+    ccsds_header : CcsdsData
+        CCSDS header data
+    """
+
+    ground_sw_version: str
+    packet_file_name: str
+    ccsds_header: CcsdsData
+
+    def parse_data(self, packet):
+        """Parse Lo L0 packet data.
+
+        Parameters
+        ----------
+        packet : dict
+            A single Lo L0 packet from space packet parser.
+
+        """
+        attributes = [field.name for field in fields(self)]
+
+        # For each item in packet, assign it to the matching attribute in the class.
+        for key, item in packet.data.items():
+            value = (
+                item.derived_value if item.derived_value is not None else item.raw_value
+            )
+            print(value)
+            if key in attributes:
+                setattr(self, key, value)
+            else:
+                raise KeyError(
+                    f"Did not find matching attribute in {self.__class__} data class"
+                    f"for {key}"
+                )
```

Comment:
You could reverse this to `if key not in attributes: raise` and avoid the else block.

Reply:
Ah good idea. I'll update that.